timelock/program/src/instruction.rs

```diff
@@ -0,0 +1,590 @@
+    ///
+    ///   0. `[writable]` Governance voting record account. See Vote docs for more detail.
+    ///   1. `[writable]` Initialized Voting account from which to remove your voting tokens.
+    ///   2. `[writable]` Initialized Yes Voting account from which to remove your voting tokens.
+    ///   3. `[writable]` Initialized No Voting account from which to remove your voting tokens.
+    ///   4. `[writable]` User token account that you wish your actual tokens to be returned to.
+    ///   5. `[writable]` Source holding account owned by the timelock that will has the actual tokens in escrow.
+    ///   6. `[writable]` Initialized Yes Voting dump account owned by timelock set to which to send your voting tokens.
+    ///   7. `[writable]` Initialized No Voting dump account owned by timelock set to which to send your voting tokens.
+    ///   8. `[writable]` Voting mint account.
+    ///   9. `[writable]` Yes Voting mint account.
+    ///   10. `[writable]` No Voting mint account.
+    ///   11. `[]` Timelock state account.
+    ///   12. `[]` Timelock set account.
+    ///   13. `[]` Transfer authority
+    ///   14. `[]` Yes Transfer authority
+    ///   15. `[]` No Transfer authority
+    ///   16. `[]` Timelock program mint authority
+    ///   17. `[]` Timelock program account pub key.
+    ///   18. `[]` Token program account.
+    WithdrawVotingTokens {
+        /// How many voting tokens to withdrawal
+        voting_token_amount: u64,
+    },
+
+    ///   0. `[writable]` Timelock config key. Needs to be set with pubkey set to PDA with seeds of the
+    ///           program account key, governance mint key, council mint key, timelock program account key.
+    ///   1. `[]` Program account that this config uses
+    ///   2. `[]` Governance mint that this config uses
+    ///   3. `[]` Council mint that this config uses [Optional] [Pass in 0s otherwise]
+    ///   4. `[]` Timelock program account pub key.
+    ///   5. `[]` Token program account.
+    InitTimelockConfig {
+        /// Consensus Algorithm
+        consensus_algorithm: u8,
+        /// Execution type
+        execution_type: u8,
+        /// Timelock Type
+        timelock_type: u8,
+        /// Voting entry rule
+        voting_entry_rule: u8,
+        /// Minimum slot time-distance from creation of proposal for an instruction to be placed
+        minimum_slot_waiting_period: u64,
+        /// Time limit in slots for proposal to be open to voting
+        time_limit: u64,
+        /// Optional name
+        name: [u8; CONFIG_NAME_LENGTH],
+    },
+
+    ///   0. `[]` Timelock config key. Needs to be set with pubkey set to PDA with seeds of the
+    ///           program account key, governance mint key, council mint key, and timelock program account key.
+    ///   1. `[]` Program account to tie this config to.
+    ///   2. `[]` Governance mint to tie this config to
+    ///   3. `[]` Council mint [optional] to tie this config to [Pass in 0s otherwise]
```

Comment:
All 0s is actually the system program, so be careful using it

Reply:
I thought all 1s was system program? What's all 1s then?

Reply:
there are no 0s in base58, so 1 is actually the smallest (and loneliest) number.  Try out https://www.appdevtools.com/base58-encoder-decoder with hex, put in a bunch of 0s on the input, and see what happens

Reply:
Oh! Well dang. The More You Know (TM) :)

Reply:
:rainbow: :star:

Reply:
Done